crates/revm/revm-inspectors/src/stack.rs

```diff
@@ -0,0 +1,240 @@
+use reth_primitives::{bytes::Bytes, Address, TxHash, H256};
+use revm::{
+    inspectors::CustomPrintTracer,
+    interpreter::{CallInputs, CreateInputs, Gas, InstructionResult, Interpreter},
+    primitives::Env,
+    Database, EVMData, Inspector,
+};
+
+/// One can hook on inspector execution in 3 ways:
+/// - Block: Hook on block execution
+/// - BlockWithIndex: Hook on block execution transaction index
+/// - Transaction: Hook on a specific transaction hash
+#[derive(Default)]
+pub enum Hook {
+    #[default]
+    /// No hook.
+    None,
+    /// Hook on a specific block.
+    Block(u64),
+    /// Hook on a specific transaction hash.
+    Transaction(TxHash),
+    /// Hooks on every transaction in a block.
+    All,
+}
+
+#[derive(Default)]
+/// An inspector that calls multiple inspectors in sequence.
+///
+/// If a call to an inspector returns a value other than [InstructionResult::Continue] (or
+/// equivalent) the remaining inspectors are not called.
+pub struct InspectorStack {
+    /// An inspector that prints the opcode traces to the console.
+    pub custom_print_tracer: Option<CustomPrintTracer>,
```

Comment:
We'll introduce other inspectors here. cc @joshieDo for snapshotting state touched